test/Microsoft.AspNetCore.Razor.Language.Test/LargeTextSourceDocumentTest.cs

```diff
@@ -6,10 +6,27 @@
 
 namespace Microsoft.AspNetCore.Razor.Language.Test
 {
-    public class LargeTextRazorSourceDocumentTest
+    public class LargeTextSourceDocumentTest
     {
         private const int ChunkTestLength = 10;
 
+        [Fact]
+        public void GetChecksum_ComputesCorrectChecksum()
+        {
+            // Arrange
+            var contentString = "Hello World";
+            var stream = TestRazorSourceDocument.CreateStreamContent(contentString);
+            var reader = new StreamReader(stream, detectEncodingFromByteOrderMarks: true);
+            var document = new LargeTextSourceDocument(reader, 5, Encoding.UTF8, "file.cshtml");
+            var expectedChecksum = new byte[] { 10, 77, 85, 168, 215, 120, 229, 2, 47, 171, 112, 25, 119, 197, 216, 64, 187, 196, 134, 208 };
+
+            // Act
+            var checksum = document.GetChecksum();
+
+            // Assert
+            Assert.Equal(expectedChecksum, checksum);
```

Comment:
> Don't we need to test cases with and without BOM?

All of our tests currently verify without BOM. The tests in this PR are verifying with BOM.

> What about other encodings?

You're right, we should be testing that checksum can handle different encodings. Today we have tests that validate the content is created properly based on encodings but not the checksum itself.